packages/fdc3-agent-proxy/src/messaging/AbstractMessaging.ts

```diff
@@ -1,165 +1,81 @@
-import { AppIdentifier, ImplementationMetadata, GetAgentParams } from '@kite9/fdc3-standard';
+import { AppIdentifier } from '@kite9/fdc3-standard';
 import { Messaging } from '../Messaging';
 import { RegisterableListener } from '../listeners/RegisterableListener';
-import {
-  AppRequestMessage,
-  WebConnectionProtocol4ValidateAppIdentity,
-  WebConnectionProtocol5ValidateAppIdentitySuccessResponse,
-} from '@kite9/fdc3-schema/generated/api/BrowserTypes';
-type RequestMetadata = AppRequestMessage['meta'];
+import { AgentResponseMessage, AppRequestMessage } from '@kite9/fdc3-schema/generated/api/BrowserTypes';
 
 export abstract class AbstractMessaging implements Messaging {
-  private readonly options: GetAgentParams;
-  private readonly connectionAttemptUuid: string;
-  private readonly timeout: number;
-  private readonly actualUrl: string;
-  private appId: AppIdentifier | null = null;
-  private implementationMetadata: ImplementationMetadata | null = null;
+  // private implementationMetadata: ImplementationMetadata;
+  private appIdentifier: AppIdentifier;
 
   abstract createUUID(): string;
   abstract post(message: object): Promise<void>;
 
   abstract register(l: RegisterableListener): void;
   abstract unregister(id: string): void;
 
-  abstract createMeta(): RequestMetadata;
+  abstract createMeta(): AppRequestMessage['meta'];
 
-  constructor(options: GetAgentParams, connectionAttemptUuid: string, actualUrl: string, timeout: number = 10016) {
-    this.options = options;
-    this.connectionAttemptUuid = connectionAttemptUuid;
-    this.timeout = timeout;
-    this.actualUrl = actualUrl;
-  }
+  abstract getTimeoutMs(): number;
 
-  getSource(): AppIdentifier {
-    if (this.appId === null) {
-      throw new Error('Unconnected. No source available.');
-    }
-    return this.appId;
+  constructor(appIdentifier: AppIdentifier) {
+    this.appIdentifier = appIdentifier;
   }
 
-  waitFor<X>(filter: (m: any) => boolean, timeoutErrorMessage?: string): Promise<X> {
+  waitFor<X extends AgentResponseMessage>(
+    filter: (m: AgentResponseMessage) => boolean,
+    timeoutErrorMessage?: string
+  ): Promise<X> {
     const id = this.createUUID();
     return new Promise<X>((resolve, reject) => {
       let done = false;
-      const listener: RegisterableListener = {
+      const l: RegisterableListener = {
         id,
         filter: filter,
         action: m => {
           done = true;
           this.unregister(id);
           resolve(m);
         },
-        register: async () => {
-          this.register(listener);
-        },
-        unsubscribe: async () => {
-          this.unregister(id);
-        },
-      };
+      } as RegisterableListener;
```

Comment:
Is `as RegisterableListener` necessary?

Reply:
It was due to the lack of the register and unregister functions, which don't get used here... I've put them back to remove this. However, I do find it architecturally confusing that both the listeners and AbstractMessaging have `register` and `unsubsubscribe`/`unregister` functions... It should probably be the responsibility of one or the other rather than either... Generally, the listeners own functions are just grabbing a reference to the Messaging instance and calling its register function anyway.